File: c++/matread1.cpp
/************************************************************************\

    matread1.cpp
    A demo program to read a matrix from a text file

    Created: 20170224

\************************************************************************/

#include <iostream>
#include <fstream>
#include <Matrix.hpp>

void read1(const char *infile)
//  read1: Reads a matrix from a file and prints it to stdout
{
    //using std::complex;

    std::cout << "## read1 ## infile=" <<  infile <<std::endl;
    std::ifstream inF(infile);

    Matrix<double> A(readmatrix1<double>(inF));
    //auto A = readmatrix1<double>(inF);
    std::cout << "dimensions = " <<  A.numrows() << " " << A.numcols() <<std::endl;

    printmatrix1(A, std::cout);
}


int main(int argc, const char **argv)
{
    if (argc < 2)
    {
        std::cerr << "Needs an input file name on arg1" << std::endl;
        return 1;
    }
    read1(argv[1]);
    return 0;
}

File: python/matread1.py
"""
matread1.py
A demo program for reading in a matrix then printing it out
"""

from __future__ import print_function
import sys
import numpy
from np_helper import loadmatrix1, printmatrix1

def read1(argv):
    if len(argv) < 2:
        print("Needs an input file name on arg1", file=sys.stderr)
        sys.exit(1)

    matfile = argv[1]
    M = loadmatrix1(matfile)
    #printmatrix1(M, float_fmt=" %9.6g")
    printmatrix1(M)

if __name__ == "__main__":
    read1(sys.argv)
